common/src/main/java/com/microsoft/identity/common/internal/providers/microsoft/azureactivedirectory/AzureActiveDirectoryAuthorizationRequest.java

```diff
@@ -22,34 +22,258 @@
+import com.microsoft.identity.common.internal.providers.oauth2.PKCEChallenge;
 
+import java.io.UnsupportedEncodingException;
 import java.net.URL;
+import java.net.URLEncoder;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.UUID;
 
 public class AzureActiveDirectoryAuthorizationRequest extends AuthorizationRequest {
+    protected static final String ENCODING_UTF8 = "UTF_8";
+    private static final String TAG = StringExtensions.class.getSimpleName();
+    private static final String RESOURCE = "resource";
+    private static final String LOGIN_HINT = "login_hint";
+    private static final String ADAL_ID_PLATFORM = "x-client-SKU";
+    private static final String ADAL_ID_VERSION = "x-client-Ver";
+    private static final String ADAL_ID_OS_VER = "x-client-OS";
+    private static final String ADAL_ID_DM = "x-client-DM";
+    private static final String CLIENT_REQUEST_ID = "client-request-id";
+    private static final String QUERY_PROMPT = "prompt";
+    private static final String CODE_CHALLENGE = "code_challenge";
+    private static final String CODE_CHALLENGE_METHOD = "code_challenge_method";
+    private static final String SCOPE_OPENID_VALUE = "openid";
+    private static final String ADAL_ID_PLATFORM_VALUE = "Android";
+    private static final String QUERY_PROMPT_VALUE = "login";
+    private static final String QUERY_PROMPT_REFRESH_SESSION_VALUE = "refresh_session";
 
+    /**
+     * Required.
+     */
     private URL mAuthority;
+    /**
+     * Required.
+     *
+     * Passed in from ADAL/MSAL after authority verification.
+     */
+    private String mAuthorizationEndpoint;
+    /**
+     * The App ID URI of the target web API.
+     * This is required in one of either the authorization or token requests.
+     * To ensure fewer authentication prompts place it in the authorization request to
+     * ensure consent is received from the user.
+     */
+    private String mResource;
+    //TODO The microsoft doc is different with V1 has currently.
+    /**
+     * Optional. Indicate the type of user interaction that is required.
+     */
+    private AADPromptBehavior mPromptBehavior;
+    /**
+     * Optional. Can be used to pre-fill the username/email address field of the sign-in page for the user, if you know their username ahead of time.
+     */
+    private String mLoginHint;
+    private UUID mCorrelationId;
+    private String mExtraQP;
+    private String mClaimsChallenge;
+    private PKCEChallenge mPKCEChallenge;
+    private String mLibraryVersion;
 
     /**
      * @return URL authority
      */
+    public String getAuthorizationEndpoint() {
+        return mAuthorizationEndpoint;
+    }
+
+    /**
+     * @param authorizationEndpoint String
+     */
+    public void setAuthorizationEndpoint(String authorizationEndpoint) {
+        mAuthorizationEndpoint = authorizationEndpoint;
+    }
+
+    public void setPKCEChallenge(final PKCEChallenge pKCEChallenge) {
```

Comment:
In sticking with the treat-acronyms-as-words thing, let's call this `setPkceChallenge`